gcc/rust/ast/rust-ast.h

```diff
@@ -788,6 +771,42 @@ class DelimTokenTree : public TokenTree, public AttrInput
   {
     return AttrInput::AttrInputType::TOKEN_TREE;
   }
+
+  std::vector<std::unique_ptr<TokenTree> > &get_token_trees ()
+  {
+    return token_trees;
+  }
+
+  DelimType get_delim_type () const { return delim_type; }
+
+  static TokenId left_delim_type_tok_token_id (DelimType delim_type)
+  {
+    switch (delim_type)
+      {
+      case PARENS:
+	return LEFT_PAREN;
+      case SQUARE:
+	return LEFT_SQUARE;
+      case CURLY:
+	return LEFT_CURLY;
+      default:
+	gcc_unreachable ();
+      }
+  }
+
+  static TokenId right_delim_type_tok_token_id (DelimType delim_type)
+  {
+    switch (delim_type)
+      {
+      case PARENS:
+	return RIGHT_PAREN;
+      case SQUARE:
+	return RIGHT_SQUARE;
+      case CURLY:
+	return RIGHT_CURLY;
+      }
+    return RIGHT_PAREN;
```

Comment:
Oops, same here. I didn't put a suggestion here in my first review
```suggestion
    switch (delim_type)
      {
      case PARENS:
	return RIGHT_PAREN;
      case SQUARE:
	return RIGHT_SQUARE;
      case CURLY:
	return RIGHT_CURLY;
      default:
        gcc_unreachable ();
      }
```

Reply:
Otherwise nice :D I'm looking forward to playing with macros in gccrs